packages/medusa/src/interfaces/file-service.ts

```diff
@@ -0,0 +1,78 @@
+export interface IFileService {
+  /**
+   * upload file to fileservice
+   * @param file Multer file from express multipart/form-data
+   * */
+  upload(file: Express.Multer.File): void
+
+  /**
+   * remove file from fileservice
+   * @param file Remove file described by record
+   * */
+  delete(file: Record<string, any>): void
+
+  /**
+   * upload file to fileservice from stream
+   * @param fileData file metadata relevant for fileservice to create and upload the file
+   * @param file readable stream of the file to upload
+   * */
+  uploadStream(
+    fileData: Record<string, any>,
+    file: NodeJS.ReadableStream
```

Comment:
File will probably be the req it self isn’t it? Since it implements the readableStream interface

Reply:
I intentionally kept it a bit vague in order to not tie it strongly to the express request such that we should be able to use the upload for exports as well, where the stream is populated in the worker process and is not a request object.

Reply:
yeah 😄